### DISCUSSION TITLE
Unsupported value for Docker version via regex manager

### DISCUSSION THREAD

**Discussion by astellingwerf** [renovatebot/renovate#10316] [Request Help] [answered]
For a Docker base image that's declared via an optional build-arg, I've configured the following regex manager:
```typescript
{
  fileMatch: [
	'(^|/)Dockerfile$'
  ],
  matchStrings: [
	'\\bARG .*_BASE_IMAGE=(?<depName>[^:]*):(?<currentValue>.*)(?:@(?<currentDigest>sha256:[a-f0-9]{64}))+?[\\r\\n\\s]',
	'\\bARG .*_BASE_IMAGE=(?<depName>[^:]*):(?<currentValue>[^@\\s]*)[\\r\\n\\s]',
  ],
  datasourceTemplate: 'docker',
  versioning: 'docker',
},
```

With a Dockerfile
```Dockerfile
ARG JAVA_BASE_IMAGE=artifactory.[REDACTED]/docker-hardened/openjdk11:11.0.10_9.202104.205186-jre-rootless
FROM $JAVA_BASE_IMAGE
```
the dependency is found, but when enabling debug logging, it is logged that the dependency is supposedly invalid:
```
Dependency artifactory.[REDACTED]/docker-hardened/openjdk11 has unsupported value 11.0.10_9.202104.205186-jre-rootless
```

When I check out the same version that's installed (25.36.6), and add a test to `lib/versioning/docker/index.spec.ts`, it succeeds.
```typescript
      expect(docker.isValid('11.0.10_9.202104.205186-jre-rootless')).toBeTruthy();
```

What's wrong with the identified version?

**Comment by rarkins** [comment, marked as answer]
Can you create a minimal reproduction repo for it so we can see the details?

**Reply by astellingwerf** [reply]
It currently relies on an internal artifactory repo, so that would take me a while...

**Reply by rarkins** [reply]
I don't think the registry needs to be available in order to reach the same failure point as in your logs though? It can even be a made-up hostname too if it's failing before the lookup.

**Reply by astellingwerf** [reply]
I'll try it later (it's weekend now!). Can you confirm that I can set the `versioning` configuration for a RegexManager like I do now?

**Reply by rarkins** [reply]
Actually, that should be `versioningTemplate`.

**Reply by astellingwerf** [reply]
Ah, that fixed it for me. I was expecting though that the type of `versioning` would have been implied from the datasource. It doesn't work that way?

**Reply by rarkins** [reply]
We should make it that way. Regex manager defaults to semver versioning so then the datasource default does not take effect. We should remove the regex manager versioning default.

**Reply by HonkingGoose** [reply]
> We should make it that way. Regex manager defaults to semver versioning so then the datasource default does not take effect. We should remove the regex manager versioning default.

Created issue #10320 to track "remove regex manager versioning default".
